### PR TITLE
feat(cdn): add cds as cdn backend + migration of cds logs

### PR DESCRIPTION
**Pull request by sguiheux**
- [x] Step logs
- [x] Services logs

@ovh/cds

### CODE REVIEW

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#14291.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Docs -   
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔ 
* Package and Analyze
  * Docker ✔ 
  * Sonar ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✘  
  * Standalone tests 1 -   
  * Standalone tests 2 -   
  * Standalone tests 3 -

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#14299.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Docs -   
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#14303.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Docs -   
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔ 
* Package and Analyze
  * Docker ✔ 
  * Sonar ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✘  
  * Standalone tests 1 -   
  * Standalone tests 2 -   
  * Standalone tests 3 -

File: engine/api/workflow_run.go [hunk 1]
```diff
@@ -35,6 +35,24 @@ const (
...
+		key := vars["permProjectKey"]
+
+		p, err := project.Load(ctx, api.mustDB(), key, project.LoadOptions.WithWorkflowNames)
+		if err != nil {
+			return err
+		}
+
+		ids, err := workflow.LoadNodeRunIDs(api.mustDB(), p.WorkflowNames.IDs())
+		if err != nil {
+			return nil
```
Comment (richardlt) [inline engine/api/workflow_run.go:50]:
```suggestion
			return err
```

File: engine/cdn/cdn_item.go [hunk 2]
```diff
@@ -34,45 +30,52 @@ func (s *Service) completeItem(ctx context.Context, itemUnit storage.ItemUnit) e
...
-	item.MD5 = md5Hash
+	item.Hash = sha512S
+	item.MD5 = md5S
 	item.Size = size
 
 	if err := index.UpdateItem(ctx, s.Mapper, tx, item); err != nil {
 		return err
 	}
 
-	if err := tx.Commit(); err != nil {
```
Comment (richardlt) [inline engine/cdn/cdn_item.go:74]: why removing ?

File: engine/cdn/cdn_sync.go [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+		if err := s.syncNodeRun(ctx, cdsStorage, pKey, nodeRunIdentifier); err != nil {
+			hasFailed = true
+			nodeRunFailed++
+			log.Error(ctx, "cdn:cds:sync:log: unable to sync node runs: %v", err)
+			continue
+		}
+		nodeRunDone++
+	}
+	log.Info(ctx, "cdn:cds:sync:log: node run done for project %s:  %d/%d (+%d failed)", pKey, nodeRunDone, len(nodeRunIds), nodeRunFailed)
+	if hasFailed {
```
Comment (richardlt) [inline engine/cdn/cdn_sync.go:77]:
```suggestion
	if nodeRunFailed > 0 {
```

**Review by richardlt — APPROVED** [review]

File: engine/cdn/cdn_item.go (hunk 2 above)
Comment (sguiheux) [reply engine/cdn/cdn_item.go:74]: transaction is no more open by this method, so this method does not need commit anymore

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#14312.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✘  
  * Docs -   
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#14313.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Docs -   
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔ 
* Package and Analyze
  * Docker ✔ 
  * Sonar ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✘  
  * Standalone tests 1 -   
  * Standalone tests 2 -   
  * Standalone tests 3 -

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#14313.1 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Docs -   
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔ 
* Package and Analyze
  * Docker ✔ 
  * Sonar ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✘  
  * Standalone tests 1 -   
  * Standalone tests 2 -   
  * Standalone tests 3 -

**Review by yesnault — APPROVED** [review]

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#14321.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Docs -   
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔

**Pull request merged**